### PR TITLE
First

### PR DESCRIPTION
**Pull request by tyson-marks29**

### CODE REVIEW

File: add_book.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+include 'database.php';
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $title = $_POST['title'];
+    $author = $_POST['author'];
+    $year = $_POST['year'];
```
Comment (tyson-marks29) [inline add_book.php:7]: To enhance readability, encapsulate database operations in separate classes and use dependency injection to manage connections more effectively.

File: add_book.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    $query = "INSERT INTO books (title, author, year) VALUES ('$title', '$author', $year)";
```
Comment (tyson-marks29) [inline add_book.php:9]: For modern PHP applications, consider the use of PDO instead of `mysqli` to take advantage of more robust database interactions and named parameters for prepared statements.

**Comment by tyson-marks29** [comment]
It's a good practice to add a new line at the end of each file to avoid potential issues with version control systems and tools processing the files.

File: delete_book.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?php
+include 'database.php';
+
+if (isset($_GET['id'])) {
```
Comment (tyson-marks29) [inline delete_book.php:4]: This code directly uses user input to construct the SQL query, making it vulnerable to SQL injection. Use prepared statements with parameter binding instead to address this security concern.

File: edit_book.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+if (isset($_GET['id'])) {
+    $id = $_GET['id'];
+    $query = "SELECT * FROM books WHERE id = $id";
+    $result = mysqli_query($connection, $query);
+    $book = mysqli_fetch_assoc($result);
+}
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $id = $_POST['id'];
+    $title = $_POST['title'];
```
Comment (tyson-marks29) [inline edit_book.php:13]: The code is directly using form inputs without validation, which may lead to unwanted bugs or inconsistent data. Validate each input to ensure that it meets expected criteria before processing.

Comment (tyson-marks29) [inline edit_book.php:13]: The application logic is intertwined with HTML view logic, which violates the Single Responsibility Principle. Extract the processing logic into a separate class or controller to adhere to this principle.

File: list_books.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?php
+include 'database.php';
+
+$query = "SELECT * FROM books";
```
Comment (tyson-marks29) [inline list_books.php:4]: The database querying design pattern is not evident in the code. Introduce a repository or a model to abstract the database interactions to improve testability and maintainability.

File: list_books.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+$result = mysqli_query($connection, $query);
+?>
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Book List</title>
+</head>
```
Comment (tyson-marks29) [inline list_books.php:11]: Consider adding unit tests for data retrieval, data modification, and input validation processes. Use a testing framework like PHPUnit to ensure that the functions work as expected and handle edge cases efficiently.
